File: SDK/PHP-SDK/ApiTubeClient.php
<?php

class ApiTubeClient
{
	private $config;

	public function __construct(ApiTubeConfig $config)
	{
		$this->config = $config;
	}

	public function request($endpoint, $params = [])
	{
		// Build query string
		$query = http_build_query($params);

		// Create the URL
		$url = $this->config->getBaseUrl() . $endpoint . '?' . $query;

		// Initialize cURL
		$request = curl_init($url);

		// Set headers
		curl_setopt($request, CURLOPT_HTTPHEADER, [
			'X-Api-Key: ' . $this->config->getApiKey(),
		]);

		// Set options to return the result
		curl_setopt($request, CURLOPT_RETURNTRANSFER, true);

		// Execute the request
		$result = curl_exec($request);

		// Close the request
		curl_close($request);

		// Decode the result
		return json_decode($result, true);
	}
}

File: SDK/Java-SDK/ApiTubeClient.java
// ApiTubeClient.java

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;
import java.net.URL;
import java.util.Map;

public class ApiTubeClient {
    private ApiTubeConfig config;

    public ApiTubeClient(ApiTubeConfig config) {
        this.config = config;
    }

    public String request(String endpoint, Map<String, String> params) throws Exception {
        // Build the query string
        StringBuilder query = new StringBuilder();
        if (params != null) {
            for (Map.Entry<String, String> entry : params.entrySet()) {
                query.append(entry.getKey())
                     .append("=")
                     .append(entry.getValue())
                     .append("&");
            }
        }

        // Build the full URL
        String urlString = config.getBaseUrl() + endpoint + "?" + query.toString();
        URL url = new URL(urlString);

        // Create the connection
        HttpURLConnection conn = (HttpURLConnection) url.openConnection();
        conn.setRequestMethod("GET");

        // Set the headers
        conn.setRequestProperty("X-Api-Key", config.getApiKey());

        // Get the response
        BufferedReader in = new BufferedReader(new InputStreamReader(conn.getInputStream()));
        String inputLine;
        StringBuilder content = new StringBuilder();
        while ((inputLine = in.readLine()) != null) {
            content.append(inputLine);
        }

        // Clean up
        in.close();
        conn.disconnect();

        return content.toString();
    }
}
